### PR TITLE
docs: add examples for operations

### LINKED ISSUE #248: docs for operations on diffraction objects
docs for functions in #187, once that issue is finished

### PR DESCRIPTION
**Pull request by yucongalicechen**
closes #248
@sbillinge ready for review

### CODE REVIEW

File: doc/source/examples/diffraction_objects_example.rst [hunk 1]
```diff
@@ -145,15 +145,38 @@ DiffractionObject convenience functions
...
-    doubled_object = 2 * diff_object1     # Double the intensities
-    sum_object = diff_object1 + diff_object2    # Sum the intensities
-    subtract_scaled = diff_object1 - 5 * diff_object2   # subtract 5 * obj2 from obj 1
+    increased_intensity = diff_object1 + 5      # Increases the intensities by 5
+    decreased_intensity = diff_object1 - 1      # Decreases the intensities by 1
+    doubled_object = 2 * diff_object1           # Double the intensities
+    reduced_intensity = diff_object1 / 2        # Halves the intensities
 
-4) get the value of the DiffractionObject at a given point in one of the xarrays
+You can also do binary operations between two diffraction objects, as long as their yarrays have the same length.
```
Comment (sbillinge) [inline doc/source/examples/diffraction_objects_example.rst:159]: is the condition "same length xarray" or "on the same `q/tth/d-array`"?  I think it should be the latter or we are adding things incorrectly.  Please can you check the code.  If the code is doing it right, change the doc.  If hte doc is right, we probably should revisit hte tests and code.

**Review by sbillinge — COMMENTED** [review]
this is great!  Just one comment.

File: doc/source/examples/diffraction_objects_example.rst (hunk 1 above)
Comment (yucongalicechen) [reply doc/source/examples/diffraction_objects_example.rst:159]: Yeah I think the code only checks for the length of yarrays. I can edit the test on this PR.

Comment (yucongalicechen) [reply doc/source/examples/diffraction_objects_example.rst]: Actually I wanna talk to @bobleesj first before making any changes. His test cases seem to be carefully chosen.. Made this into a new issue #298. @sbillinge I changed the docs and it's ready for review

**Comment by sbillinge** [comment]
Thanks @yucongalicechen LGTM

**Pull request merged**